app/src/main/java/com/android/partagix/ui/App.kt

```diff
@@ -162,20 +177,29 @@ class App(
 
   override fun onSignInSuccess(user: FirebaseUser?) {
     if (user != null) {
-      val newUser =
-          User(
-              user.uid,
-              user.displayName ?: "",
-              "Unknown Location",
-              "0",
-              Inventory(user.uid, emptyList()))
-      db.getUser(user.uid, { db.createUser(newUser) }, {})
-    }
-    // test that navigationActions has been initialized
+      notificationManager.checkToken(user.uid) { newToken ->
+        val newUser =
+            User(
+                user.uid,
+                user.displayName ?: "",
+                user.email ?: "",
```

Comment:
Here, you need to leave the "Unknown Location field" or we will still have the Location as the email address``
```suggestion
                "Unknown Location",
```